spec.html

```diff
@@ -4393,6 +4393,20 @@ <h1>The Lexical Environment and Environment Record Specification Types</h1>
       <p>The Lexical Environment and Environment Record types are used to explain the behaviour of name resolution in nested functions and blocks. These types and the operations upon them are defined in <emu-xref href="#sec-lexical-environments"></emu-xref>.</p>
     </emu-clause>
 
+    <emu-clause id="sec-abstract-closure">
+      <h1>The Abstract Closure Specification Type</h1>
+      <p>The <dfn>abstract closure</dfn> specification type is used to refer to algorithmic steps together with a collection of values. Abstract closures are meta-values and are invoked using function application style such as _closure_(_arg1_, _arg2_). Like abstract operations, invocations perform the algorithmic steps described by the abstract closure. Unlike abstract operations, abstract closures are meta-values and must be created.</p>
```

Comment:
It is perhaps worth calling out that Abstract Closures return completion records, perhaps even completion records with `[[Type]]` `normal` or `throw` (as we do for [internal methods](https://tc39.es/ecma262/#sec-invariants-of-the-essential-internal-methods)).

Reply:
This is mentioned above as https://github.com/tc39/ecma262/pull/1889#pullrequestreview-369293586 :wink:

Reply:
> Unlike abstract operations, abstract closures are meta-values and must be created.

What does "must be created" possibly mean in regard to "meta-values" which " do not necessarily correspond to any specific entity within an ECMAScript implementation".  Like all other specification devices, Abstract Closures are exclusively intended  to define required semantics, not actual implementation techniques.  "created" sounds like implementation and its explicit use in describing Abstract Closures could be construed to imply that they are different in that regard from other   meta-values such as Completion Records.

Reply:
Basically I mean there must be prose like "let _foo_ be a new abstract closure [...]". This is consistent with other meta-values like "let _l_ be a new empty List" and so forth, and contrasts with abstract operations which just... are. AOs aren't meta-values but just collections of algorithmic steps.

Reply:
@allenwb:

I take it to mean that referring to an abstract closure requires a previous step which says "Let _c_ be a new abstract closure [...]" or similar, whereas abstract operations conceptually "exist" at all points in time and can be referred to without such a step.

Would you be happier if this said "... and must be introduced before they can be referred to"?

Reply:
How is your suggested statement different for Abstract Closures from any other kind of meta-value?  Doesn't a List or Completion Record also have to "be introduced" before for it can be referred to?

Reply:
@allenwb Fair enough. This clarification was more important in a previous iteration of the PR that had a declarative named form of introducing abstract closures. In the current iteration there is only inline ways to introduce abstract closures, so this sentence could be removed without much loss.

Reply:
This sentence was removed.